src/Spectre.libPeakFinder.Tests/ExtremaFinderTest.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * ExtremaFinderTest.cpp
+ * Tests whether peaks and valleys of signal are properly being found.
+ *
+ Copyright 2018 Michal Gallus
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#include <gtest/gtest.h>
+#include "Spectre.libPeakFinder\ExtremaFinder.h"
+
+namespace
+{
+    using namespace spectre::algorithm::peakfinder;
+
+    TEST(ExtremaFinderInitialization, initializes)
+    {
+        EXPECT_NO_THROW(ExtremaFinder());
+    }
+
+    class ExtremaFinderTest : public ::testing::Test
+    {
+    public:
+        ExtremaFinderTest()
+        {
+            signal = {
+                1.0, 2.0, 2.5, 2.8, 3.5, 2.8, 4.0, 3.0,
+                3.0, 1.0, 1.5, 2.0, 2.5, 2.0, 1.0, 0.0
```

Comment:
Interesting case. Good that it's not the most straightforward one :)

Reply:
Thanks! I've used https://plot.ly/create/ to visualise the data during creation.